### PR TITLE
Feature/3646 legacy js removal data api

### PR DESCRIPTION
**Pull request by ivankruchkoff**
## Summary

Addresses issue #3646

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by ivankruchkoff** [comment]
@googlebot I consent.

File: tests/e2e/specs/modules/optimize/activation.test.js [hunk 1]
```diff
@@ -79,20 +79,7 @@ describe( 'Optimize Activation', () => {
...
-					.url()
-					.match(
-						'/wp-json/google-site-kit/v1/modules/analytics/data/report?'
-					)
-			) {
-				request.respond( {
-					status: 200,
-					body: JSON.stringify( { placeholder_response: true } ),
-				} );
-			} else if (
```
Comment (eugene-manuilov) [inline tests/e2e/specs/modules/optimize/activation.test.js:95]: Looks like this has been deleted by accident. Let's revert this.

File: tests/e2e/specs/modules/pagespeed-insights/activation.test.js [hunk 2]
```diff
@@ -18,17 +18,7 @@ describe( 'PageSpeed Insights Activation', () => {
...
-				request.respond( { status: 200 } );
-			} else if (
-				request
-					.url()
-					.match(
-						'google-site-kit/v1/modules/pagespeed-insights/data/pagespeed'
-					)
-			) {
-				request.respond( { status: 200, body: JSON.stringify( {} ) } );
-			} else if (
```
Comment (eugene-manuilov) [inline tests/e2e/specs/modules/pagespeed-insights/activation.test.js:31]: Same here. This should remain unchanged. Please, add it back.

File: assets/js/modules/adsense/index.js [hunk 3]
```diff
@@ -19,22 +19,19 @@
...
  * WordPress dependencies
  */
-import { addFilter } from '@wordpress/hooks';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import { AREA_DASHBOARD_EARNINGS } from '../../googlesitekit/widgets/default-areas';
-import { fillFilterWithComponent } from '../../util';
```
Comment (eugene-manuilov) [inline assets/js/modules/adsense/index.js:29]: After removing this, the `fillFilterWithComponent` function is not needed anymore and we can get rid of it too. Please, remove it too.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @ivankruchkoff. This mostly looks good to me. Added three comments for you. Once they are addressed, it will be good to merge.

**Comment by tofumatt** [comment]
@googlebot I consent.

File: tests/e2e/specs/modules/optimize/activation.test.js (hunk 1 above)
Comment (ivankruchkoff) [reply tests/e2e/specs/modules/optimize/activation.test.js:95]: It was in the commit linked in the IB https://github.com/google/site-kit-wp/commit/d77d4fc8d20ff6d5fa5e266c9be495c3bc3d8ae3

> * [Remove all E2E request inceptions for unused `v1/data` API calls](https://github.com/google/site-kit-wp/commit/d77d4fc8d20ff6d5fa5e266c9be495c3bc3d8ae3)

File: tests/e2e/specs/modules/pagespeed-insights/activation.test.js (hunk 2 above)
Comment (ivankruchkoff) [reply tests/e2e/specs/modules/pagespeed-insights/activation.test.js:31]: It was in the commit linked in the IB https://github.com/google/site-kit-wp/commit/d77d4fc8d20ff6d5fa5e266c9be495c3bc3d8ae3

> * [Remove all E2E request inceptions for unused `v1/data` API calls](https://github.com/google/site-kit-wp/commit/d77d4fc8d20ff6d5fa5e266c9be495c3bc3d8ae3)

File: assets/js/modules/adsense/index.js (hunk 3 above)
Comment (ivankruchkoff) [reply assets/js/modules/adsense/index.js:29]: https://github.com/google/site-kit-wp/pull/3823/commits/4a48393b2f9c98c5dc3adb63e35e5d6b13f5b081

File: tests/e2e/specs/modules/pagespeed-insights/activation.test.js (hunk 2 above)
Comment (ivankruchkoff) [reply tests/e2e/specs/modules/pagespeed-insights/activation.test.js]: Re-added it in https://github.com/google/site-kit-wp/pull/3823/commits/9c273d6fe57af3e5b1779779bc64eb19b81eab08

File: tests/e2e/specs/modules/optimize/activation.test.js (hunk 1 above)
Comment (ivankruchkoff) [reply tests/e2e/specs/modules/optimize/activation.test.js]: Re-added it in https://github.com/google/site-kit-wp/pull/3823/commits/9c273d6fe57af3e5b1779779bc64eb19b81eab08

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
In general, looks good to me, but there are E2E issues and we need to re-remove request interceptions for `/v1/data` datapoints in the following files:

* tests/e2e/specs/modules/optimize/activation.test.js
* tests/e2e/specs/modules/pagespeed-insights/activation.test.js

File: tests/e2e/specs/modules/optimize/activation.test.js (hunk 1 above)
Comment (eugene-manuilov) [reply tests/e2e/specs/modules/optimize/activation.test.js]: @ivankruchkoff you also added request interception for `v1/data` endpoint. We don't need it, so it should be re-removed.

**Review by eugene-manuilov — APPROVED** [review]
LGTM

**Pull request merged**